src/librustdoc/html/render/print_item.rs

```diff
@@ -1041,7 +1044,13 @@ fn item_enum(w: &mut Buffer, cx: &Context<'_>, it: &clean::Item, e: &clean::Enum
             document_non_exhaustive(w, variant);
 
             use crate::clean::Variant;
-            if let clean::VariantItem(Variant::Struct(ref s)) = *variant.kind {
+            if let Some((extra, fields)) = match *variant.kind {
+                clean::VariantItem(Variant::Struct(ref s)) => {
+                    Some((if s.struct_type == CtorKind::Fn { "Tuple " } else { "" }, &s.fields))
```

Comment:
This `if` condition will always be false, right? Can you remove the `if`?

Reply:
I guess, I'll add an `unreachable`, just in case.

Reply:
It's better to simply move it actually.